pkg/route/hostassignment/assignment.go

```diff
@@ -86,41 +92,61 @@ func AllocateHost(ctx context.Context, route *routev1.Route, sarc SubjectAccessR
 	return nil
 }
 
-func hasCertificateInfo(tls *routev1.TLSConfig) bool {
+func hasCertificateInfo(tls *routev1.TLSConfig, opts route.RouteValidationOptions) bool {
 	if tls == nil {
 		return false
 	}
-	return len(tls.Certificate) > 0 ||
+	hasInfo := len(tls.Certificate) > 0 ||
 		len(tls.Key) > 0 ||
 		len(tls.CACertificate) > 0 ||
 		len(tls.DestinationCACertificate) > 0
+
+	if opts.AllowExternalCertificates && tls.ExternalCertificate != nil {
+		hasInfo = hasInfo || len(tls.ExternalCertificate.Name) > 0
+	}
+	return hasInfo
 }
 
-func certificateChangeRequiresAuth(route, older *routev1.Route) bool {
+func certificateChangeRequiresAuth(route, older *routev1.Route, opts route.RouteValidationOptions) bool {
 	switch {
 	case route.Spec.TLS != nil && older.Spec.TLS != nil:
 		a, b := route.Spec.TLS, older.Spec.TLS
-		if !hasCertificateInfo(a) {
+		if !hasCertificateInfo(a, opts) {
 			// removing certificate info is allowed
 			return false
 		}
-		return a.CACertificate != b.CACertificate ||
+
+		certChanged := a.CACertificate != b.CACertificate ||
 			a.Certificate != b.Certificate ||
 			a.DestinationCACertificate != b.DestinationCACertificate ||
 			a.Key != b.Key
+
+		if opts.AllowExternalCertificates && route.Spec.TLS != nil && older.Spec.TLS != nil {
+			if route.Spec.TLS.ExternalCertificate != nil || older.Spec.TLS.ExternalCertificate != nil {
+				certChanged = true
```

Comment:
Because we cannot definitively verify if the content of the secret that is referenced has been modified.

The reason this also mentioned in the EP:

If the old route or the new route uses `.spec.tls.externalCertificate` this validation will always have the precondition [certificateChangeRequiresAuth()](https://github.com/openshift/library-go/blob/d8d3f3f8a9e4a82c110a89a13229ce1412a88e4a/pkg/route/hostassignment/assignment.go#L123C29-L123C29) return true since we cannot definitively verify if the content of the secret that is referenced has been modified. Since the previous validation func (ValidateHostExternalCertificate()) would have already validation user permissions, we can safely make this assumption.

Reply:
I was wondering abou the secret name, not the secret contents.

Reply:
Can the external certificate secret name change?

Reply:
Yeah! The external certificate secret name can be changed, but the caveat here is even if the secret name remains the same, we need to assume that the secret content is changed, and hence, the certificate is changed.

Reply:
I do see you added a comment for this caveat to `ValidateHostUpdate`, but perhaps it deserves a code comment in `certificateChangeRequiresAuth` as well.